src/CourseDashboard.cpp

```diff
@@ -2,31 +2,62 @@
 #include "FileHandler.hpp"
 #include "Utility.hpp"
 #include "AuthenticationProvider.hpp"
+#include <iostream>
 
 
-void CourseDashboard::loadFromFile(const std::string& pathTofile)
+void CourseDashboard::loadFromJson(const std::string& pathToFile)
```

Comment:
I would keep loadFromFile and later if required use design pattern (e.g. factory for different format file readers) by user